src/pudl/transform/ferc1.py

```diff
@@ -769,10 +769,8 @@ def _plants_steam_clean(ferc1_steam_df):
             "expns_kwh": 'opex_per_kwh'})
         .pipe(_clean_cols, "f1_steam")
         .pipe(pudl.helpers.simplify_strings, ['plant_name_ferc1'])
-        .pipe(pudl.helpers.cleanstrings,
-              ['construction_type', 'plant_type'],
-              [CONSTRUCTION_TYPE_STRINGS, PLANT_KIND_STRINGS],
-              unmapped='')
+        .pipe(pudl.helpers.cleanstrings, ['construction_type'], [CONSTRUCTION_TYPE_STRINGS], unmapped=pd.NA)
+        .pipe(pudl.helpers.cleanstrings, ['plant_type'], [PLANT_KIND_STRINGS], unmapped="")
```

Comment:
`plant_type` wasn't listed as a column that needed to remove empty strings in the issue, so when I encountered this I skipped `plant_type`. Also a few tests failed when I changed it to pd.NA (I forget what the error is but could regenerate), maybe the column is treated as a primary key? I could go back and change this to NAs though

Reply:
In `fields.py` there's this comment: 

TODO Disambiguate column name and apply table specific ENUM constraints. There are different allowable values in different tables.

So it seems like maybe this is a good candidate to remove the empty string?

Reply:
Ah okay. I don't think it's ever used as a primary key. There's a mechanism for overriding the generic purely field-name based attributes if we need to. It looks like `plant_type` shows up in in the `plants_hydro_ferc1`, `plants_small_ferc1`, and `plants_steam_ferc1` tables, but currently only the `plants_steam_ferc1` version is being cleaned and set to some canonical values defined in the `PLANT_KIND_STRINGS` , so to do what the TODO asks would require categorizing all the (different) nonsense strings in the hydro and small plants tables. I don't know if this is something that might be getting done already in @aesharpe's FERC 1 cleaning PR but we should ask.

I think the thing to do for now if we want to apply a table specific ENUM is to override the constraints which are defined for this column only for the `plants_steam_ferc1` table, using the `FIELD_METADATA_BY_RESOURCE` defined at the bottom of the `fields.py` module (currently there's only one override). And then maybe create an issue to remind us to create ENUM constraints for the `plant_type` field in the hydro and small plants tables in concert with Austen's cleanup of those tables.

Reply:
Right now, the `plant_type` column gets cleaned and standardized in the steam table, manually mapped in the small generators table, left alone in the hydro table, and is non-existent (and irrelevant) in the pumped storage table.

Part of my FERC 1 cleaning module extracts the headers from the small generators table, standardizes them, and puts them in the `plant_type` column so that we don't have to rely on manual mapping. This is still a draft PR, however, so none of this is integrated yet. 

The `plant_type` extracted from the headers is standardized according to a dictionary that I created and which we can change to reflect whatever final plant type string we want (or specify in an ENUM constraint). 

Current small plant fuel types: 
```
None, 'hydro', 'internal_combustion', 'solar_pv', 'cogen',
       'steam_heat', 'wind', 'gas_turbine', 'diesel_turbine',
       'pumped_hydro', 'coal', 'waste_heat', 'fuel_cell'
```

Current steam fuel types: 
```
'steam', 'nuclear', 'unknown', 'combustion_turbine',
       'combined_cycle', 'internal_combustion', 'geothermal', 'wind',
       'photovoltaic', 'solar_thermal', ''
```

At the very least, we could standardize the NA values for both of these tables. Ideally we will pull values for `plant_type` from the same set of values, whether they are constrained or not. I think it's a good idea to create an issue to remind us to do this (and for the hydro table and maybe even an arbitrary one for the pumped storage table so that it has a `plant_type` when it gets merged with the others into the mega FERC table). Ultimately, the goal of all this RMI manual mapping we're doing is to connect, among other things, the EIA `technology_description` field with individual FERC records so we won't need to use the janky FERC `plant_type` or header extraction at all.

Reply:
For now I've moved `plant_type` to the FIELD_METADATA_BY_RESOURCE and added an assert so that there are no uncaught strings in the `plant_type` column of `plants_steam_ferc1`. Seems like there should be a new issue that adds ENUM constraints for the FERC 1 hydro and small plants tables? Or is the goal to remove `plant_type` altogether and have it agree with EIA `technology_description`?